datamol/viz/_lasso_highlight.py

```diff
@@ -0,0 +1,439 @@
+
+    def generate_attachment_points(self):
+        sorted_bonds = sorted(self.bonds, key=lambda x: x.angle)
+        self.bond_attachment_points = dict()
+        for i, bond in enumerate(sorted_bonds):
+            alpha_left = bond.angle - self.delta_angle
+            alpha_right = bond.angle + self.delta_angle
+
+            d_left = self.radius
+            d_right = self.radius
+
+            # Handling intersecting bonds
+            # # Intersections with previous bonds
+            if i == 0:  # For first bond the previous bond is the last bond. Subtracting 2 pi.
+                prev_bond_angle = sorted_bonds[-1].angle - np.pi * 2
+            else:
+                prev_bond_angle = sorted_bonds[i - 1].angle
+
+            # #  If both points intersect the mean angle is calculated.
+            if prev_bond_angle + self.delta_angle >= alpha_left:
+                alpha_left = np.mean([prev_bond_angle + self.delta_angle, alpha_left])
+
+                a_rhombus = bond.angle - prev_bond_angle
+
+                len_rhombus = self.bond_width / np.sin(a_rhombus)
+                # Radius is altered to match the intersecting position
+                d_left = 2 * len_rhombus * np.cos(a_rhombus / 2)
+
+            # # Intersections with following bonds
+            if i + 1 == len(sorted_bonds):
+                next_bond_angle = sorted_bonds[0].angle + np.pi * 2
+            else:
+                next_bond_angle = sorted_bonds[i + 1].angle
+
+            if next_bond_angle - self.delta_angle <= alpha_right:
+                alpha_right = np.mean([next_bond_angle - self.delta_angle, alpha_right])
+
+                a_rhombus = next_bond_angle - bond.angle
+                len_rhombus = self.bond_width / np.sin(a_rhombus)
+                d_right = 2 * len_rhombus * np.cos(a_rhombus / 2)
+
+            self.bond_attachment_points[bond.bond_id] = [
+                (alpha_left, d_left),
+                (alpha_right, d_right),
+            ]
+        return self
+
+    def get_arch_attachment_points(self) -> Iterator[Tuple[float, float]]:
+        """Points between bonds which are drawn as arch."""
+
+        if self.bonds:
+            sorted_bonds = sorted(self.bonds, key=lambda x: x[0])
+            _, _, bond_keys = zip(*sorted_bonds)
+            for i, k in enumerate(bond_keys):
+                if i == 0:
+                    start_angle = self.bond_attachment_points[bond_keys[-1]][1][0] - np.pi * 2
+                else:
+                    start_angle = self.bond_attachment_points[bond_keys[i - 1]][1][0]
+                end_angle = self.bond_attachment_points[k][0][0]
+                if np.isclose(start_angle % (np.pi * 2), end_angle % (np.pi * 2)):
+                    continue
+                yield start_angle, end_angle
+
+
+ColorTuple = Union[Tuple[float, float, float, float], Tuple[float, float, float]]
+
+
+def _draw_substructurematch(
+    canvas: rdMolDraw2D.MolDraw2D,
+    mol: Chem.Mol,
```

Comment:
Replace all `Chem.Mol` by `dm.Mol`. They both point to the same type but `dm.Mol` is more typing friendly.